segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/Storage.java

```diff
@@ -172,8 +171,21 @@
      */
     boolean supportsTruncation();
 
+    /**
+     * Determines whether this Storage implementation supports atomic writes. Supporting atomic writes means that calls
+     * to {@link #write} will either make all the payload available for reading (via {@link #getStreamSegmentInfo} and
+     * {@link #read}) or none, regardless of whether the underlying Storage binding supports that (i.e., FileSystem does not).
```

Comment:
Nit: 
binding supports that (E.g., FileSystem does not)

Reply:
fixed